### ISSUE TITLE
[unnecessary-else] Add option to disable error at else-if statements

### ISSUE THREAD

**Issue by timocov** [palantir/tslint#4598]
### Feature request

**Is your feature request related to a problem? Please describe.**

It would be nice to have an option to configure [`unnecessary-else` rule](https://github.com/palantir/tslint/pull/4502) to don't fail on `else if` statements (only on `else`).

**Comment by JoshuaKGoldberg** [comment]
Hi @timocov! Could you explain more on why you want this rule option? It seems rather arbitrary to allow `else if` but not `else`.

**Comment by timocov** [comment]
For me it looks strange to change a code from

```ts
if (a > 0) {
  return 1;
} else if (a < 0) {
  return 2;
} else {
  return 3;
}
```

to

```ts
if (a > 0) {
  return 1;
}

if (a < 0) {
  return 2;
}

return 3;
```

These conditions are related and (for me) they should be in "one flow" (not separated conditions), because if someone decide to add new condition between `if (a > 0) {` and `if (a < 0) {` it "breaks" their relating.

But I agree, that `else` block in this case is unnecessary and should be "removed":

```ts
if (a > 0) {
  return 1;
} else if (a < 0) {
  return 2;
}

return 3;
```

**Comment by JoshuaKGoldberg** [comment]
Seems reasonable as a style preference. Not very important in the context of linting, but it's nice to relax these style preference rules when requested.
